zingolib/src/wallet/send.rs

```diff
@@ -213,16 +212,21 @@ impl LightWallet {
 
             let height = calculated_transaction.status().get_height();
 
+            let lrz_transaction = calculated_transaction.transaction();
+            let consensus_branch_id = lrz_transaction.consensus_branch_id();
+            tracing::debug!(
+                "Sending transaction with the following consensus BranchId: {consensus_branch_id:?}, at height {height}."
+            );
+
             let mut transaction_bytes = vec![];
-            calculated_transaction
-                .transaction()
+            lrz_transaction
                 .write(&mut transaction_bytes)
                 .map_err(|_| TransmissionError::TransactionWrite)?;
-            let transaction = Transaction::read(
-                transaction_bytes.as_slice(),
-                consensus::BranchId::for_height(&network, height),
-            )
-            .map_err(|_| TransmissionError::TransactionRead)?;
+
+            // this block serves to save the transactions to the wallet. This consensus branch is not baked into the sent transaction right here. because only transaction_bytes will be sent.
+            let transaction = Transaction::read(transaction_bytes.as_slice(), consensus_branch_id)
```

Comment:
Was `consensus::BranchId::for_height` returning a branch ID that differed from the branch ID that was calculated for the transaction we're sending? Or is this simply removing an additional source of truth? I'm in favor of using the ID we already calculated instead of regenerating it...but if we were regenerating it wrong, that's interesting to me in its own right.